Website/Providers/DataProviders/SqlDataProvider/09.02.00.SqlDataProvider

```diff
@@ -492,16 +492,87 @@ BEGIN
                                                 END AS PropertyValue
                                             FROM   {databaseOwner}{objectQualifier}vw_Users AS U
                                                 INNER JOIN {databaseOwner}{objectQualifier}UserProfile AS P ON U.UserID = P.UserID
-                                                LEFT OUTER JOIN {databaseOwner}{objectQualifier}ProfilePropertyDefinition AS Prop ON 
+                                                LEFT OUTER JOIN {databaseOwner}{objectQualifier}ProfilePropertyDefinition AS Prop ON
                                                 (Prop.PropertyDefinitionID = P.PropertyDefinitionID and Prop.Deleted = 0 and Prop.PortalID = ' + CONVERT(nvarchar(20), @PortalID) + ')
                                             WHERE U.UserId IN (SELECT UserId FROM @TempUsers) AND (U.PortalId = ' + CONVERT(nvarchar(20), @PortalID) + ' OR U.PortalId IS NULL)
                                             )' +
-                @pivotSql + @sortSql + ' DROP TABLE #MatchingUsers'           
+                @pivotSql + @sortSql + ' DROP TABLE #MatchingUsers'
 
         EXEC(@sql)
 END
 GO
 
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}AddPasswordHistory]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+	DROP PROCEDURE {databaseOwner}{objectQualifier}AddPasswordHistory
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}AddPasswordHistory]
+	@UserId             int,
+	@Password           nvarchar(128),
+	@PasswordSalt       nvarchar(128),
+	@PasswordsRetained  int,
+	@DaysRetained       int,
```

Comment:
daysRetained is not used in the SPROC, so why add it?

Reply:
it is used on line 547

Reply:
it is used on line 547